src/session/session_helper.c

```diff
@@ -134,3 +134,35 @@ __wt_session_dump(WT_SESSION_IMPL *session, WT_SESSION_IMPL *dump_session, bool
     __wt_scr_free(session, &buf);
     return (ret);
 }
+
+/*
+ * __wt_session_set_last_error --
+ *     Stores information about the last error to occur during this session.
+ */
+int
+__wt_session_set_last_error(
+  WT_SESSION *wt_session, int err, int sub_level_err, const char *err_msg_content)
+{
+    WT_DECL_RET;
+    WT_SESSION_IMPL *session = (WT_SESSION_IMPL *)wt_session;
+
+    char *err_msg = session->err_info.err_msg;
+    size_t err_msg_size;
+
+    /* Ensure arguments are valid, and that session is not a mock */
+    WT_ASSERT(session, __wt_is_valid_sub_level_error(sub_level_err));
+    if (err_msg_content == NULL || wt_session->get_last_error == NULL)
```

Comment:
`wt_session->get_last_error` I don't see a scenario where the function pointer could be NULL here, anyhow filling the struct should not be related to checking if the function was set correctly. 

I would rather remove it instead.

Reply:
It's NULL when the session is a mock, I'm not sure what I should do to handle that case.

Reply:
I don't know in which case we will use a mock session without assigning function pointer to session, can you check and see if we can modify the code where we create a mock session? 

The other thing worth mentioning here is `return (0)` indicates function executes as expected and returns "OK". We should 1. assert `err_msg_content` is not `NULL` or 2. do the check and return some error code if it's NULL. This rule also applies to the check of function pointer, but for function pointer if we can modify the code where we have the mock session to ensure it will never be `NULL`, we can remove the check directly.

Reply:
I've removed the check in [Updated mock session and altered __wt_session_set_last_error](https://github.com/wiredtiger/wiredtiger/pull/11361/commits/6c29c7c8e3950f9ff8d5b27c593f29f8bf857afc)
The original issue was that mock sessions did not free the err_info.err_msg string when closing, which lead to memory leaks, so I wanted the function to no-op for mock sessions. 
I've updated the mock sessions to free the memory in the destructor, so the check is no longer needed.